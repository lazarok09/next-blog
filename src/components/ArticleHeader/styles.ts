import styled, { css } from "styled-components";
import { Title as HeadingStyles } from "../Heading/styles";

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      @media ${theme.media.lteMedium} {
        font-size: calc(${theme.font.sizes.medium} + 1rem);
      }
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
    @media ${theme.media.lteMedium} {
      font-size: calc(${theme.font.sizes.small} + 0.2rem);
    }
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
