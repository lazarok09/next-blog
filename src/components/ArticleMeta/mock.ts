import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: 'aowea3i902103102',

    slug: 'lazaro-souza',
    displayName: 'LÃ¡zaro Souza',
  },
  categories: [
    {
      id: '123125129120',
      displayName: 'JS',
      slug: 'javascript',
    },
    {
      id: '123125129120125512',
      displayName: 'Tech',
      slug: 'tech',
    },
  ],
} as ArticleMetaProps;
