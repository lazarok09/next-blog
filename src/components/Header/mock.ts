import { HeaderProps } from '.';

export default {
  blogName: 'LÃ¡zaro',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: 'https://github.com/lazarok09.png',
  showText: true,
} as HeaderProps;
