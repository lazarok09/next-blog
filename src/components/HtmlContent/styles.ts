import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    p {
      margin: ${theme.spacings.xlarge} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      transition: all 300ms ease-in-out;
      text-decoration: none;
    }
    a:hover {
      filter: brightness(80%);
      text-decoration: underline;
    }
    pre, code{
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.xlarge} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
  }
    img {
        width: 100%;

    }
    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xlarge} 0;
      max-width: 100%;

    }
    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }
    .image-style-side {
      float: right;
      max-width: 50%;
      margin: 0 ${theme.spacings.medium};
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }
    ul,
    ol {
      margin: ${theme.spacings.xlarge};
    }
    .table {
      max-width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 100%;
      border-collapse: collapse;
    }
    table th,
    table td{
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }
    }
    iframe {
    max-width: 100%;
    height: 50vh;
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: 0;
      }
    }
  `}
`;
