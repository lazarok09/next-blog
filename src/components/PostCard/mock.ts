import { data } from "../../api/data.json";

export default data.posts[0];
