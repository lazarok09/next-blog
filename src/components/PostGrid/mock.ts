import { data } from '../../api/data.json';

export default { posts: data.posts };
