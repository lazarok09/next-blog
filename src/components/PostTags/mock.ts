import { data } from '../../api/data.json';

export default {
  tags: data.posts[0].tags,
};
