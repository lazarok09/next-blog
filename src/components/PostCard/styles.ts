import styled, { css } from "styled-components";
import { Title as HeadingStyles } from "../Heading/styles";
export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small};
    }

    & a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }
    & a:hover {
      color: ${theme.colors.secondary};
    }
    // when we hover this element & i want to change the behavier of the a tag
    &:hover a {
      color: ${theme.colors.secondary};
    }
    &:hover img {
      opacity: 0.8;
    }
  `}
`;
export const Cover = styled.img`
  ${() => css`
    max-width: 100%;
    height: 100%;
    max-height: 280px;
    transition: opacity 300ms ease-in-out;
  `}
`;
export const Excerpt = styled.p`
  ${() => css`
    word-break: break-word;
  `}
`;
