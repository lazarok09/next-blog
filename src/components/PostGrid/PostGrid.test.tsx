import { screen } from '@testing-library/react';
import PostGrid, { PostGridProps } from '.';
import { renderTheme } from '../../styles/render-theme';
import mock from './mock';

const props: PostGridProps = mock;
describe('<PostGrid />', () => {
  it('should not render posts', () => {
    renderTheme(<PostGrid {...props} posts={undefined} />);
    expect(screen.getByText(/Nenhum post/i)).toBeInTheDocument();
  });

  it('should render two posts', () => {
    const { container } = renderTheme(<PostGrid {...props} />);
    expect(screen.queryByText(/Nenhum post/i)).not.toBeInTheDocument();

    expect(screen.getAllByRole('heading')).toHaveLength(2);
    expect(screen.getAllByRole('img')).toHaveLength(2);
    expect(container.querySelectorAll('p')).toHaveLength(2);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostGrid {...props} />);
    expect(container).toMatchSnapshot();
  });
});
