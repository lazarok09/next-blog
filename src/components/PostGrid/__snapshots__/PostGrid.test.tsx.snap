// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> should match snapshot 1`] = `
.c5 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c2 .c4 {
  margin: 0;
  margin-top: calc(1.6rem - 0.6rem);
  margin-bottom: 1.6rem;
}

.c2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 a:hover {
  color: #dc143c;
}

.c2:hover a {
  color: #dc143c;
}

.c2:hover img {
  opacity: 0.8;
}

.c3 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c6 {
  word-break: break-word;
}

.c0 {
  width: 100%;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

@media (max-width:360px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          href="/post/basics-java-script-stuff"
        >
          <img
            alt="Basics JavaScript stuff"
            class="c3"
            src="https://res.cloudinary.com/lazarok09/image/upload/v1633798940/photo_1560415755_bd80d06eda60_ixid_Mnwx_Mj_A3f_DB_8_M_Hxwa_G90by1w_Y_Wdlf_Hx8f_G_Vuf_DB_8f_Hx8_and_ixlib_rb_1_2_f32360cc03.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/basics-java-script-stuff"
          >
            Basics JavaScript stuff
             
          </a>
        </h2>
        <p
          class="c6"
        >
          JavaScript is a cross-platform, object-oriented scripting language. The v8 by Chrome runs the Node and make JS be a performatic language.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/como-usar-o-context-api-do-react"
        >
          <img
            alt="Como usar o contextAPI do React"
            class="c3"
            src="https://res.cloudinary.com/lazarok09/image/upload/v1633798940/photo_1560415755_bd80d06eda60_ixid_Mnwx_Mj_A3f_DB_8_M_Hxwa_G90by1w_Y_Wdlf_Hx8f_G_Vuf_DB_8f_Hx8_and_ixlib_rb_1_2_f32360cc03.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/como-usar-o-context-api-do-react"
          >
            Como usar o contextAPI do React
             
          </a>
        </h2>
        <p
          class="c6"
        >
          Como utilizar o contextAPI do React, compartilhar dados na aplicação e mudar o estado de componentes.
        </p>
      </div>
    </div>
  </div>
</div>
`;
